engine/model/job.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"github.com/pingcap/tiflow/dm/pkg/log"
+	"go.uber.org/zap"
+)
+
+type (
+	// JobID is the unique identifier of a job
+	JobID = string
+	// JobType is the type of a type, like `CVSDEMO`, `DM`, `CDC`.
+	JobType = int32
+)
+
+// Define job type
+const (
+	JobTypeInvalid = JobType(iota)
+	// job type for job manager, only for inner use
+	JobTypeJobManager
+	JobTypeCVSDemo
+	JobTypeDM
+	JobTypeCDC
+	JobTypeFakeJob
+)
+
+var jobTypeName = map[int32]string{
```

Comment:
Usually, using a switch statement is faster than the map. See https://stackoverflow.com/questions/46789259/map-vs-switch-performance-in-go.